### PR TITLE
Check for unpinned dependencies

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

Somebody (🙈 🙋‍♂️) always seems to be forgetting to pin dependencies when opening PRs (latest proof: https://github.com/concrete-utopia/utopia/pull/4867#discussion_r1484429968).

Therefore, let's have the robots fix ~~my~~ our mistakes, for once.

**Fix:**

1. Add a pre-commit check via husky that will exit with a non-0 code if there are unpinned deps in any of the folders listed as targets in `check-unpinned-deps.js`
2. Add a CI step that runs on PRs that does the same thing **if for some reason the pre-commit hook did not do its job**.

**Note:** I commented out the checks on `devDependencies`, because I'm not sure whether we should enforce it there or not.

### CODE REVIEW

**Review by bkrmendy — APPROVED** [review]

File: check-unpinned-deps.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const fs = require('fs')
+const { join } = require('path')
+
+/**
+ * Scan dependencies in the package.json files in these folders.
+ */
+const folders = ['.', './editor', './utopia-remix', './puppeteer-tests']
```
Comment (Rheeseyb) [inline check-unpinned-deps.js:7]: Can you add `utopia-api`, `utopia-vscode-common`, and `utopia-vscode-extension` to this list please

**Review by Rheeseyb — APPROVED** [review]
Thank you :)

File: check-unpinned-deps.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const unpinnedDeps = folders.flatMap((projectPath) => {
+  const path = join(projectPath, 'package.json')
+  const packageJson = JSON.parse(fs.readFileSync(path), 'utf8')
+  const allDeps = {
+    ...packageJson.dependencies,
+    // ...packageJson.devDependencies,
+  }
+  return Object.entries(allDeps)
+    .filter(([, version]) => version.match(/^[\^~]/))
```
Comment (seanparsons) [inline check-unpinned-deps.js:17]: Not sure I understand what the regex is checking, can we not just do `version.startsWith('^')`?

Comment (ruggi) [reply check-unpinned-deps.js]: yeah it's pointlessly unreadable, updated in https://github.com/concrete-utopia/utopia/pull/4870/commits/190d1ae2c4170ff0fa1b1f24592c782c7af82a82

**Review by liady — APPROVED** [review]

**Review by Rheeseyb — APPROVED** [review]

**Pull request merged**